lte/gateway/python/magma/pipelined/app/classifier.py

```diff
@@ -277,100 +313,189 @@ def delete_tunnel_flows(self, i_teid:int, ue_ip_adr:IPAddress,
 
             flows.delete_flow(self._datapath, self.tbl_num, match)
 
-        # Delete flow for LOCAL port
-        if not ue_ip_adr:
-            self.logger.error("ue_ip_address is None")
-            return
+        if ip_flow_dl.set_params:
+            self._delete_tunnel_ip_flow_dl(ip_flow_dl)
         else:
-            ip_match_out = get_ue_ip_match_args(ue_ip_adr, Direction.IN)
-            match = MagmaMatch(eth_type=get_eth_type(ue_ip_adr),
-                               in_port=self._uplink_port, **ip_match_out)
-            flows.delete_flow(self._datapath, self.tbl_num, match)
-
-            # Delete flow for mtr port
-            match = MagmaMatch(eth_type=get_eth_type(ue_ip_adr),
-                               in_port=self.config.mtr_port, **ip_match_out)
+            # Delete flow for LOCAL port
+            if not ue_ip_adr:
```

Comment:
we need API that generate MagmaMatch from given parameters. This API needs to be called from add and delete APIs. This would be simplify code a bit and avoid code duplication.